src/Preprocessor.zig

```diff
@@ -1232,6 +1232,43 @@ fn expandFuncMacro(
                     some.expansionSlice(),
                 ) else try pp.pragmaOperator(string.?, loc);
             },
+            .comma => {
+                if (tok_i + 2 < func_macro.tokens.len) {
+                    const maybe_hashhash = func_macro.tokens[tok_i + 1];
+                    const maybe_va_args = func_macro.tokens[tok_i + 2];
+                    if (maybe_hashhash.id == .hash_hash and maybe_va_args.id == .keyword_va_args) {
+                        // GNU extension: `, ##__VA_ARGS__` deletes the comma if __VA_ARGS__ is empty
+                        tok_i += 2;
+                        if (func_macro.params.len == expanded_args.items.len) {
+                            // Empty __VA_ARGS__, drop the comma
+                            try pp.err(maybe_hashhash, .comma_deletion_va_args);
+                        } else if (func_macro.params.len == 0 and expanded_args.items.len == 1 and expanded_args.items[0].len == 0) {
+                            // Ambiguous whether this is "empty __VA_ARGS__" or "__VA_ARGS__ omitted"
```

Comment:
The ambiguity is in the preprocessor specification, not the aro implementation. e.g. if this is valid:
```c
#define ONE_ARG(X) X + 1
ONE_ARG();
```
Because X has zero tokens. So applying that to the VA_ARGS case:
```c
#define FOO(...) foo(a, ##__VA_ARGS__)
FOO();
```
It's not clear, syntactically, if `FOO` is being "called" with one zero-token parameter (meaning `__VA_ARGS__` is present, so keep the comma, and put those zero tokens after it) or if it's being called with *no* parameters, which means `__VA_ARGS__` is empty, so the comma should be deleted. Clang and GCC decide which behavior to use based on whether a GNU standard is being used or not (GNU standard will delete the comma, C standard will not).